tests/scripts/analyze_outcomes.py

```diff
@@ -493,10 +489,19 @@ def main():
 
         KNOWN_TASKS['analyze_coverage']['args']['full_coverage'] = options.full_coverage
 
+        # If the outcome file already exists, we assume that the user wants to
+        # perform the comparison.
+        # Share the contents among tasks to improve performance.
+        if os.path.exists(options.outcomes):
+            main_results.info("Read outcome file from {}.", options.outcomes)
+            outcomes_or_file = read_outcome_file(options.outcomes)
```

Comment:
I'm not sure about `outcomes_or_file` arguments. This relates to a partially pre-existing issue: generating the outcomes file on the fly only works when there's a single task and it's a `driver_vs_reference` tasks (as opposed to `analyze_coverage`).

I'm thinking perhaps we should move outcomes file generation to here instead:
```
if not os.path.exists(options.outcomes):
    if len(task_list) == 1 and KNOWN_TASKS[task_list[0]]['test_function'] == do_analyze_driver_vs_reference:
        # generate outcomes files by calling execute_reference_driver_tests()
    else:
        # this is an error

outcomes = read_outcome_file(options.outcomes)
```
That way (1) we give an clean error when the outcomes files doesn't exist and generating it won't work, and (2) all functions can now accept a parse `outcomes` instead of doing the `isinstance` dance.

Wdyt?

Reply:
As we are improving this, I think we could go further. How about adding `prepare_outcomes_function` for each task in `KNOWN_TASKS`， and call this task specific function for each task before the `test_function` is called.